src/Microsoft.AspNet.Mvc.TagHelpers/ScriptTagHelper.cs

```diff
@@ -51,83 +125,166 @@ public class ScriptTagHelper : TagHelper
 
+            if (!modeResult.FullMatches.Any())
+            {
+                // No attributes matched so we have nothing to do
                 return;
             }
 
-            var content = new StringBuilder();
+            // Get the highest matched mode
+            var mode = modeResult.FullMatches.Select(match => match.Mode).Distinct().Max();
 
-            // NOTE: Values in Output.Attributes are already HtmlEncoded
+            // NOTE: Values in TagHelperOutput.Attributes are already HtmlEncoded
+            var attributes = new Dictionary<string, string>(output.Attributes);
+            
+            var builder = new StringBuilder();
+            var originalContent = await context.GetChildContentAsync();
 
-            // We've taken over rendering here so prevent the element rendering the outer tag
-            output.TagName = null;
+            if (mode == Mode.Fallback && string.IsNullOrEmpty(SrcInclude))
+            {
+                // No globbing to do, just build a <script /> tag to match the original one in the source file
+                BuildScriptTag(originalContent, attributes, builder);
+            }
+            else
+            {
+                BuildGlobbedScriptTags(originalContent, attributes, builder);
+            }
 
-            // Rebuild the <script /> tag.
-            content.Append("<script");
-            foreach (var attribute in output.Attributes)
+            if (mode == Mode.Fallback)
             {
-                content.AppendFormat(CultureInfo.InvariantCulture, " {0}=\"{1}\"", attribute.Key, attribute.Value);
+                BuildFallbackBlock(attributes, builder);
             }
 
-            content.Append(">");
+            // We've taken over tag rendering, so prevent rendering the outer tag
+            output.TagName = null;
+            output.Content = builder.ToString();
+        }
 
-            var originalContent = await context.GetChildContentAsync();
-            content.Append(originalContent)
-                   .AppendLine("</script>");
+        private void BuildGlobbedScriptTags(
+            string originalContent,
+            IDictionary<string, string> attributes,
+            StringBuilder builder)
+        {
+            // Build a <script> tag for each matched src as well as the original one in the source file
+            string staticSrc;
+            attributes.TryGetValue("src", out staticSrc);
 
-            // Build the <script> tag that checks the test method and if it fails, renders the extra script.
-            content.Append("<script>(")
-                   .Append(FallbackTestExpression)
-                   .Append("||document.write(\"<script");
+            EnsureGlobbingUrlBuilder();
+            var srcs = GlobbingUrlBuilder.BuildUrlList(staticSrc, SrcInclude, SrcExclude);
 
-            if (!output.Attributes.ContainsKey("src"))
+            foreach (var src in srcs)
```

Comment:
Seems odd to expand `src` to `sources` when I'm intending to refer to the semantic meaning of the `src` attribute. But no big deal.